app/main.py

```diff
@@ -1,3 +1,57 @@
-def shop_trip():
-    # write your code here
-    pass
+import json
+
+from app.shop import Shop
+from app.customer import Customer
+
+
+def shop_trip() -> None:
+    with open("app/config.json", "r") as config_file:
+        config = json.load(config_file)
+
+    customers_data = config["customers"]
+    shops_data = config["shops"]
+    fuel_price = config["FUEL_PRICE"]
+
+    customers = [Customer(**customer_data) for customer_data in customers_data]
+    shops = [Shop(**shop_data) for shop_data in shops_data]
+
+    for customer in customers:
+        print(f"{customer.name} has {customer.money} dollars")
+
+        cheapest_shop = None
+        min_cost = float("inf")
+
+        for shop in shops:
+            cost_fuel_for_road = customer.calculate_trip_cost(
+                shop.location, fuel_price)
```

Comment:
In case your open parenthesis does not have arguments on the same line, your closing parenthesis should not have arguments on the same line as well.